pdp/ast/yast/selector.go

```diff
@@ -23,42 +23,21 @@ func (ctx context) unmarshalSelector(v interface{}) (pdp.LocalSelector, boundErr
 		return pdp.LocalSelector{}, newSelectorURIError(s, ierr)
 	}
 
-	if strings.ToLower(ID.Scheme) == "local" {
-		loc := strings.Split(ID.Opaque, "/")
+	scheme := strings.ToLower(ID.Scheme)
+	loc := strings.Split(ID.Opaque, "/")
+
+	if scheme == "local" {
 		if len(loc) != 2 {
 			return pdp.LocalSelector{}, newSelectorLocationError(ID.Opaque, s)
 		}
 
-		items, err := ctx.extractList(m, yastTagPath, "path")
-		if err != nil {
-			return pdp.LocalSelector{}, bindErrorf(err, "selector(%s.%s)", loc[0], loc[1])
-		}
-
-		path := make([]pdp.Expression, len(items))
-		for i, item := range items {
-			e, err := ctx.unmarshalExpression(item)
-			if err != nil {
-				return pdp.LocalSelector{}, bindErrorf(bindErrorf(err, "%d", i), "selector(%s.%s)", loc[0], loc[1])
-			}
-
-			path[i] = e
-		}
-
-		s, err := ctx.extractString(m, yastTagType, "type")
-		if err != nil {
-			return pdp.LocalSelector{}, bindErrorf(err, "selector(%s.%s)", loc[0], loc[1])
-		}
-
-		t, ok := pdp.TypeIDs[strings.ToLower(s)]
-		if !ok {
-			return pdp.LocalSelector{}, bindErrorf(newUnknownTypeError(s), "selector(%s.%s)", loc[0], loc[1])
-		}
-
-		if t == pdp.TypeUndefined {
-			return pdp.LocalSelector{}, bindErrorf(newInvalidTypeError(t), "selector(%s.%s)", loc[0], loc[1])
+		return makeLocalSelector(ctx, loc, m)
+	} else if scheme == "pip" {
+		if len(loc) != 2 {
```

Comment:
I'd make validation for len(loc) > 0 here. The only PDP wants to know name of pip. Rest part of string can be treated differently by particular PIP implementation.